class2.md

```diff
@@ -0,0 +1,287 @@
+
+*Overfitting* occurs when a model is fitted too closely to the specific artifacts of a dataset. In this situation the model will work very well with training data but fail to generalize to new data.
+
+*Underfitting* occurs when a model is too simple to capture variability in the dataset. In this situation the model won't work well on the training set.
+* Example: "Everyone who smokes will get cancer" does not capture any of the variablity in possible outcomes.
+
+### Bias-variance trade off
+
+<p align="center">
+  <img width="600" alt="" src="images/modelcomplexity.png">
+</p>
+
+As a model gets more complex, it will become better at predicting on training data. However, if it becomes too complex then it will begin to fail to generalize to new data. This relationship is called the *bias-variance trade off*.
+
+* Bias is the amount of error introduced by approximating real-world phenomena with a simplified model.
+* Variance is how much your model's test error changes based on variation in the training data. It reflects the model's sensitivity to the quirks of the dataset it was trained on.
+
+The ideal situation is one where performance is both highly accurate and generalizable to new data. As you can see in the image above on the training set increasing model complexity will result in a more accurate result. However, on the testing set increasing model complexity only works to improve accuracy to an extent after which it become detrimental.
+
+#### The relationship between model complexity and dataset size
+
+Model complexity is closely tied to the variation of inputs contained in the testing set. Variation can only be comprehensively captured by having very large datasets. You can avoid overfitting by increasing the size of the datasets and therefore increasing the variety of data points in your dataset.
+
+**An important note is that adding data points that are duplicates or very similar to those already captured in the dataset will not increase complexity**
+
+### Two kinds of supervised learning: classification and regression
+
+The main difference between classification and regression is that classification predicts a categorical variable and regression predicts a continuous numerical value.
+
+<p align="center">
+  <img width="350" alt="" src="images/supervised.png">
+</p>
+
+### Classification
+
+Classification aims to predict a class label, which is a choice from a predetermined list of possibilities.
+* Binary classification: Classifying into exactly two classes.
+* Multiclass classification: Classifying into more than two classes.
+
+*Decision trees* and *support vector machines* are two kinds of classification methods.
```

Comment:
these are different from each other in <these ways> (or define each), focusing now on decision trees, because....